pyvista/core/composite.py

```diff
@@ -460,23 +463,40 @@ def __setitem__(self, index: Union[Tuple[int, Optional[str]], int, str], data: D
             i, name = cast(int, index), None
         if data is not None and not is_pyvista_dataset(data):
             data = wrap(data)
+
         if i == -1:
             self.append(data)
             i = self.n_blocks - 1
         else:
+            # this is the only spot in the class where we actually add
+            # data to the MultiBlock
+
+            # check if we are overwriting a block
+            existing_dataset = self.GetBlock(i)
+            if existing_dataset is not None:
```

Comment:
The reason for this guarding is unclear to me.  I would expect `_refs` to mirror the datasets.  If there is a `None`, I would expect to also keep track of this in `refs` somehow.

Reply:
This is here in the rare edge case where the dataset no longer exists within the multiblock because it was removed by accessing the underlying VTK API with `RemoveBlock`. You're correct, normally the underlying `_refs` mirrors the state of this object.

I think it's ok to leave it in here, but I'm open to suggestions.